### PR TITLE
check the db connection before registering the permissions 

### PR DESCRIPTION
**Pull request by MerQ-ai**
## Description

checking the database connection before the registering the permissions instead of Defer registering permissions on the Gate instance until it's resolved to grant the permissions before other authorization checks 
## Motivation and context

to be able to use @can directive properly 

## How has this been tested?

this is a refactor ( don't think any test needed ) 

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply.

Please, please, please, don't send your pull request until all of the boxes are ticked. Once your pull request is created, it will trigger a build on our [continuous integration](http://www.phptherightway.com/#continuous-integration) server to make sure your [tests and code style pass](https://help.github.com/articles/about-required-status-checks/).

- [x] I have read the **[CONTRIBUTING](CONTRIBUTING.md)** document.
- [x] My pull request addresses exactly one patch/feature.
- [ ] I have created a branch for this patch/feature.
- [x] Each individual commit in the pull request is meaningful.
- [ ] I have added tests to cover my changes.
- [x] If my change requires a change to the documentation, I have updated it accordingly.

### CODE REVIEW

**Comment by mostafamaklad** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/PermissionServiceProvider.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mostafamaklad/laravel-permission-mongodb/pullRequest?prid=5194487&bid=17036679)

File: src/PermissionServiceProvider.php [hunk 1]
```diff
@@ -41,9 +42,12 @@ public function boot()
 
         $this->registerModelBindings();
 
-        $this->app->afterResolving(Gate::class, function () {
+        try {
+            DB::connection()->getPdo();
             app(PermissionRegistrar::class)->registerPermissions();
-        });
+        } catch (\Exception $e) {
+            \Log::error("Could not connect to the database.  Please check your configuration. error:" . $e);
```
Comment (mostafamaklad) [inline src/PermissionServiceProvider.php]: Codacy Issue found: [Avoid using static access to class '\Log' in method 'boot'.](https://app.codacy.com/manual/mostafamaklad/laravel-permission-mongodb/pullRequest?prid=5194487)

**Comment by mostafamaklad** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/PermissionServiceProvider.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mostafamaklad/laravel-permission-mongodb/pullRequest?prid=5194487&bid=17036679)

File: src/PermissionServiceProvider.php [hunk 2]
```diff
@@ -41,9 +41,12 @@ public function boot()
 
         $this->registerModelBindings();
 
-        $this->app->afterResolving(Gate::class, function () {
+        try {
+            DB::connection()->getPdo();
             app(PermissionRegistrar::class)->registerPermissions();
-        });
+        } catch (\Exception $e) {
+            Log::error("Could not connect to the database.  Please check your configuration. error:" . $e);
```
Comment (mostafamaklad) [inline src/PermissionServiceProvider.php]: Codacy Issue found: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'boot'.](https://app.codacy.com/manual/mostafamaklad/laravel-permission-mongodb/pullRequest?prid=5194487)

**Comment by mostafamaklad** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/PermissionServiceProvider.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mostafamaklad/laravel-permission-mongodb/pullRequest?prid=5194487&bid=17036679)

File: src/PermissionServiceProvider.php (hunk 2 above)
Comment (mostafamaklad) [inline src/PermissionServiceProvider.php]: Codacy Issue found: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'boot'.](https://app.codacy.com/manual/mostafamaklad/laravel-permission-mongodb/pullRequest?prid=5194487)

**Review by MerQ-ai — COMMENTED** [review]
fixed the issues and pushed to master

File: src/PermissionServiceProvider.php (hunk 2 above)
Comment (MerQ-ai) [reply src/PermissionServiceProvider.php]: fixed

**Comment by MerQ-ai** [comment]
fixed the issue , please check

**Comment by mostafamaklad** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/PermissionServiceProvider.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mostafamaklad/laravel-permission-mongodb/pullRequest?prid=5194487&bid=17036679)

**Comment by mostafamaklad** [comment]
Thanks for your contribution 👍

**Pull request merged**